src/battle_script_commands.c

```diff
@@ -16150,11 +16147,22 @@ void BS_CalcMetalBurstDmg(void)
     }
 }
 
+void BS_JumpIfCantFling(void)
+{
+    NATIVE_ARGS(u8 battler, const u8 *jumpInstr);
+
+    u32 battler = GetBattlerForBattleScript(cmd->battler);
+    if (!CanFling(battler))
+        gBattlescriptCurrInstr = cmd->jumpInstr;
+    else
+        gBattlescriptCurrInstr = cmd->nextInstr;
+}
+
```

Comment:
You're already receiving the battler from `NATIVE_ARGS` here, so there shouldn't be a need to call `GetBattlerForBattleScript`, and the data type change for `CanFling`.

I think some natives have that issue, so that's something to clean up. (eg, compare `BS_GetBattlerSide` vs `BS_CanTeleport`)

```suggestion
void BS_JumpIfCantFling(void)
{
    NATIVE_ARGS(u8 battler, const u8 *jumpInstr);

    if (!CanFling(cmd->battler))
        gBattlescriptCurrInstr = cmd->jumpInstr;
    else
        gBattlescriptCurrInstr = cmd->nextInstr;
}
```

Reply:
Actually, no, I'm wrong! `BS_CanTeleport` is literally the only one that ISN'T using that, so that's a problem there.

Reply:
Nope, this is needed. The battler in the args tells which battler to use.

Reply:
I know, that I said in my second comment ^^
Though `GetBattlerForBattleScript` returns `u8`, so there was still no need to change `CanFling`'s argument data type.